strawberryfields/parameters.py

```diff
@@ -19,335 +19,334 @@
-        return self._wrap(temp + other)
-
-    def __radd__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other + temp)
-
-    def __sub__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp - other)
-
-    def __rsub__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other - temp)
-
-    def __mul__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp * other)
-
-    def __rmul__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other * temp)
-
-    def __truediv__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp / other)
-
-    def __rtruediv__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other / temp)
-
-    def __pow__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp ** other)
-
-    def __rpow__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other ** temp)
-
-    def __neg__(self):
-        # pylint: disable=invalid-unary-operand-type
-        return Parameter(-self.x)
-
-    # other properties
-    @property
-    def shape(self):
-        """Returns the shape of array parameters."""
-        try:
-            return self.x.shape
-        except AttributeError:
-            return None
-
-    # comparisons
-    def __eq__(self, other):
-        """Equality comparison.
-
-        .. note:: This method may be too permissive, maybe it should return False if either parameter is not a numbers.Number or a np.ndarray?
+    Used for deferred measurements, i.e., using a measurement's value
+    symbolically in defining a gate before the numeric value of that
+    measurement is available.
+
+    Former RegRefTransform functionality is provided by the sympy.Symbol base class.
+
+    Args:
+        regref (RegRef): register reference responsible for storing the measurement result
+    """
+
+    def __new__(cls, regref):
+        # sympy.Basic.__new__ wants a name, other arguments must not end up in self._args
+        return super().__new__(cls, 'q'+str(regref.ind))
+
+    def __init__(self, regref):
+        if not regref.active:
+            # TODO: Maybe we want to delete a mode right after measurement to save comp effort?
+            # The measurement result would still be preserved in the RegRef...
+            raise ValueError('Trying to use an inactive RegRef.')
+        #: RegRef: the parameter value depends on this RegRef, it can only be evaluated after the corresponding subsystem has been measured
+        self.regref = regref
+
+    def _sympystr(self, printer):
+        """"The Sympy printing system uses this method instead of __str__."""
+        return str(self.regref) + '.par'
+
+    def _eval_evalf(self, prec):
```

Comment:
convention from `sympy`? Irksome name

Reply:
Yeah. `sympy.Symbol.evalf()` eventually calls this method. Technically we do not have to use it since we call `_eval_evalf` directly, so I can change the name if you like.